### PR TITLE
USHIFT-3124: Update RHEL-9.4 AMIs to GA image from beta images

### PR DESCRIPTION
**Pull request by copejon**
Jobs should use RHEL-9.4 releases now that they are available as AMIs.

[USHIFT-3124](https://issues.redhat.com//browse/USHIFT-3124)

### CODE REVIEW

File: ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh [hunk 1]
```diff
@@ -20,7 +20,8 @@ declare -A ami_map=(
   [us-west-2,x86_64,rhel-9.4]=ami-0c0178316edfbe8d4    # RHEL-9.4.0_HVM_BETA-20240411-x86_64-30-Hourly2-GP3
   [us-west-2,arm64,rhel-9.2]=ami-0cb125bb261a63f52     # RHEL-9.2.0_HVM-20240229-arm64-33-Hourly2-GP3
   [us-west-2,arm64,rhel-9.3]=ami-04379fa947a959c92     # RHEL-9.3.0_HVM-20240229-arm64-27-Hourly2-GP3
-  [us-west-2,arm64,rhel-9.4]=ami-04f55670af52c6a38     # RHEL-9.4.0_HVM_BETA-20240411-arm64-30-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.4]=ami-036d52dc86a04b099     # RHEL-9.4.0_HVM-20240423-arm64-62-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.4]=ami-0a04d3220939627d1     # RHEL-9.4.0_HVM-20240423-x86_64-62-Hourly2-GP3
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh:24]: Should ths be x86_64, replacing the fist line?

File: ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh [hunk 2]
```diff
@@ -20,7 +20,8 @@ declare -A ami_map=(
   [us-west-2,x86_64,rhel-9.4]=ami-0c0178316edfbe8d4    # RHEL-9.4.0_HVM_BETA-20240411-x86_64-30-Hourly2-GP3
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh:20]: Is this line still required?

File: ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh [hunk 3]
```diff
@@ -20,7 +20,8 @@ declare -A ami_map=(
...
   [us-west-2,arm64,rhel-9.2]=ami-0cb125bb261a63f52     # RHEL-9.2.0_HVM-20240229-arm64-33-Hourly2-GP3
   [us-west-2,arm64,rhel-9.3]=ami-04379fa947a959c92     # RHEL-9.3.0_HVM-20240229-arm64-27-Hourly2-GP3
-  [us-west-2,arm64,rhel-9.4]=ami-04f55670af52c6a38     # RHEL-9.4.0_HVM_BETA-20240411-arm64-30-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.4]=ami-036d52dc86a04b099     # RHEL-9.4.0_HVM-20240423-arm64-62-Hourly2-GP3
+  [us-west-2,x86_64,rhel-9.4]=ami-0a04d3220939627d1     # RHEL-9.4.0_HVM-20240423-x86_64-62-Hourly2-GP3
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh:24]: The sorting in this block is per-OS and per-platform. Can we keep this?

File: ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh (hunk 2 above)
Comment (copejon) [reply ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh:20]: Ah, missed that it was 9.4

**Comment by ggiguash** [comment]
/pj-rehearse ci/rehearse/openshift/microshift/main/microshift-metal-tests ci/rehearse/openshift/microshift/main/microshift-metal-tests-arm

**Comment by ggiguash** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests-arm pull-ci-openshift-microshift-main-microshift-metal-tests

**Comment by ggiguash** [comment]
/retitle USHIFT-3124: Update RHEL-9.4 AMIs to GA image from beta images

**Pull request merged**